### PR TITLE
Add eigen to package.xml and alphabetize the entries.

### PR DESCRIPTION
**Pull request by mpowelson**

### CODE REVIEW

File: tesseract/tesseract_collision/package.xml [hunk 1]
```diff
@@ -9,7 +9,9 @@
 
   <depend>tesseract_geometry</depend>
   <depend>tesseract_common</depend>
+
   <depend>bullet3</depend>
+  <depend>eigen</depend>
```
Comment (gavanderhoorn) [inline tesseract/tesseract_collision/package.xml]: Seeing as Eigen is a header-only library, shouldn't this be a `build_depend` and `build_export_depend` instead?

Comment (mpowelson) [reply tesseract/tesseract_collision/package.xml]: I suppose. What does that practically get you? I can replace them with build_export_depend if you want, but I'd rather not have to go through all of the packages and see how it is being used. Currently I just added it if there was a find_package in the cmake.

Comment (gavanderhoorn) [reply tesseract/tesseract_collision/package.xml]: More fine grained dependency management.

Eigen is not a large library, but it's not very nice to install packages which are build dependencies only on systems that will never be compiling something.

`depend` == `union(build_depend, build_export_depend, exec_depend)`.

And on systems where Eigen is not actually distributed as a package (and those do exist), this saves users from having to `skip-key` yet another key a maintainer has just blanketly added to the `depend` set.

**Comment by gavanderhoorn** [comment]
Thanks for changing it, but `eigen` needs to be both a `build_depend` *and* a `build_export_depend` (ie: it's needed for the current build, and it's needed by packages that use your packages, when they are built).

**Pull request merged**